src/analysis/processing/qgsalgorithmzonalstatisticsfeaturebased.cpp

```diff
@@ -0,0 +1,189 @@
+/***************************************************************************
+                         qgsalgorithmzonalstatisticsfeaturebased.cpp
+                         ---------------------
+    begin                : September 2020
+    copyright            : (C) 2020 by Matthias Kuhn
+    email                : matthias@opengis.ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmzonalstatisticsfeaturebased.h"
+
+///@cond PRIVATE
+
+const std::vector< QgsZonalStatistics::Statistic > STATS
+{
+  QgsZonalStatistics::Count,
+  QgsZonalStatistics::Sum,
+  QgsZonalStatistics::Mean,
+  QgsZonalStatistics::Median,
+  QgsZonalStatistics::StDev,
+  QgsZonalStatistics::Min,
+  QgsZonalStatistics::Max,
+  QgsZonalStatistics::Range,
+  QgsZonalStatistics::Minority,
+  QgsZonalStatistics::Majority,
+  QgsZonalStatistics::Variety,
+  QgsZonalStatistics::Variance,
+};
+
+QString QgsZonalStatisticsFeatureBasedAlgorithm::name() const
+{
+  return QStringLiteral( "zonalstatisticsfb" );
+}
+
+QString QgsZonalStatisticsFeatureBasedAlgorithm::displayName() const
+{
+  return QObject::tr( "Zonal statistics (feature based)" );
```

Comment:
```suggestion
  return QObject::tr( "Zonal statistics" );
```

Lets just deprecate the old one -- I really don't want two algorithms showing in the toolbox for this one function, it's way too confusing for users!